domain/externalcontroller/state/state.go

```diff
@@ -90,28 +90,23 @@ FROM   external_controller AS ctrl
        ON        ctrl.uuid = model.controller_uuid
        LEFT JOIN external_controller_address AS addrs
        ON        ctrl.uuid = addrs.controller_uuid
-WHERE  ctrl.uuid = (
+WHERE  ctrl.uuid IN (
     SELECT controller_uuid
     FROM   external_model AS model
-    WHERE  model.uuid = $M.model
+    WHERE  model.uuid IN ($ModelUUIDs[:])
 )`
 
-	s, err := st.Prepare(q, Controller{}, sqlair.M{})
+	s, err := st.Prepare(q, Controller{}, dbModelUUIDs)
 	if err != nil {
 		return nil, errors.Annotatef(err, "preparing %q", q)
 	}
 
 	var resultControllerInfos Controllers
 	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		for _, modelUUID := range modelUUIDs {
-			var rows Controllers
-			err := tx.Query(ctx, s, sqlair.M{"model": modelUUID}).GetAll(&rows)
-			if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
-				return errors.Trace(domain.CoerceError(err))
-			}
-			resultControllerInfos = append(resultControllerInfos, rows...)
+		err := tx.Query(ctx, s, dbModelUUIDs).GetAll(&resultControllerInfos)
```

Comment:
**suggestion:** Use one letter name for receiver. Here we should use `statement`, especially since the receiver is a `State` (which is name `st` for some reason, but it could have been `s` :D)

Reply:
Changed, though I've made it `stmt` for consistency because that's what we use everywhere else